### DISCUSSION TITLE
Computer crashing when pushing or pulling to remmote repo

### DISCUSSION THREAD

**Discussion by Caluben** [community/community#137469] [Repositories] [answered]
### Select Topic Area

Question

### Body

I am not sure best practice for these forums so forgive me if I don't give all the information that I should, as I don't know what's needed.

I am using Arch Linux with 16GB of ram. 

Recently I have been trying to set up a remote repository to have version control and easy transfer of my dotfiles for things like neovim, tmux, i3, alacritty etc. This is my first time working with GitHub. I successfully set up my ssh key and repository up until the point of pushing my commits to the remote repo. Upon doing so nothing happens and my ram usage slowly goes up until it reaches 100%, at which point my computer freezes up and I have to shutdown in order to fix it. I initially tried adding all of the files in a single commit and pushing that. I tried again adding one directory, committing it, and pushing it one at a time. Nothing changed it just crashed slower. Does anyone know where I should even start when fixing a problem like this? Anything at all is greatly appreciated.

**Comment by oguzhan18** [comment, marked as answer]
Hey @Caluben,

This sounds like a frustrating issue, but we can break it down and troubleshoot it step by step. Based on your description, it seems like the system is running out of resources (specifically RAM) during the git push process, which could indicate a few things.

Potential Causes & Fixes:

1. Large Commit or Large Files:

- If you're pushing a very large commit (or if there are large files like binaries, images, etc.), Git might be trying to compress these files, which could cause high memory usage.
- Fix: Try checking the size of the files you're committing. You can list all large files in the repository with:
```bash
git rev-list --objects --all | sort -k 2 > allfiles.txt
```

If you see large files, you might want to consider using [Git LFS](https://git-lfs.github.com/) for handling large files efficiently:
```bash
git lfs install
git lfs track "*.file_extension"
```
Then, push again and see if the issue persists.

2.  SSH Config or Network Issues:
- Since you mentioned setting up SSH, it’s worth double-checking your SSH configuration to ensure it's not causing connection problems that could lead to the system hanging.
- Fix: Try testing your SSH connection manually with
```bash
ssh -T git@github.com
```
If there’s any lag or delay here, your SSH config might need tweaking, or you might want to switch to HTTPS for the repository temporarily to isolate whether SSH is the issue:
```bash
git remote set-url origin https://github.com/username/repository.git
````
3. Memory Leak or High Resource Usage:

- Arch Linux is known for its flexibility, but sometimes processes can unexpectedly consume all available memory (RAM). Git’s operations, especially with large repositories or many files, might cause a high load.
- Fix: You can monitor the RAM usage while pushing using htop or free -m to see if there's a particular process or bottleneck during the push. If Git is the cause, try committing smaller chunks and pushing incrementally:
```bash
git add .
git commit -m "Adding only smaller changes"
git push
```
4. Repository Corruption:
- In rare cases, your local repository might be in a corrupt state, causing excessive memory usage when interacting with the remote.
- Fix: You can try cloning a fresh copy of the repository and pushing from there:
```bash
git clone --mirror <repo-url>
```
5. Git Configuration:

- Sometimes memory issues can arise due to the way Git is configured to handle objects and diffs.
- Fix: You can try setting Git to use less memory by tweaking its internal buffer size:
```bash
git config --global core.compression 1
GIT_TRACE_PACKET=1 GIT_TRACE=1 GIT_CURL_VERBOSE=1 git push origin master
```

**Reply by Caluben** [reply]
I'm pretty sure it was an ssh configuration issue. I tried cloning a repo yesterday using ssh and ran into the same issue. I cloned the same repo using https and it worked right away. I am still trying to isolate the problem causing this issue but it is definitely my ssh configuration messing with me. Thank you for the help!
